### DISCUSSION TITLE
Pipeline failed: An error occurred (SignatureDoesNotMatch) when calling the AssumeRole operation

### DISCUSSION THREAD

**Discussion by gberlot-luna** [gruntwork-io/knowledge-base#733] [How do I]
We just finished to configure pipeline using the following documentation: https://github.com/gruntwork-clients/infrastructure-live-luna-solutions-llc-2/blob/main/docs/03-configure-gw-pipelines.md (We used  the example that's on the documentation to trigger it)

however the following error was displayed.

2023-06-07 20:52:57 [INFO] [aws-auth] Assuming role arn:aws:iam::XXXXXXXXX:role/allow-auto-deploy-from-other-accounts. These creds will expire after 3600 seconds.

An error occurred (SignatureDoesNotMatch) when calling the AssumeRole operation: The request signature we calculated does not match the signature you provided. Check your AWS Secret Access Key and signing method. Consult the service documentation for details.
Error: Process completed with exit code 123.

Please advice, how to fix it.

---

<ins datetime="2023-06-07T21:00:44Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110240">Tracked in ticket #110240</a></p>
</ins>

**Comment by gitsstewart** [comment]
@gberlot-luna it sounds like the error may be related to not having MFA set up correctly? Does this KB post help solve your issue https://github.com/orgs/gruntwork-io/discussions/6

**Reply by gberlot-luna** [reply]
The solution was, delete existing aws-keys and re-create new ones. (MFA should be created first)
